src/github.com/getlantern/pac-cmd/linux.c

```diff
@@ -26,17 +27,26 @@ int togglePac(bool turnOn, const char* pacUrl)
     }
   }
   else {
+    if (strlen(pacUrl) != 0) {
+      // clear pac setting only if it's equal to pacUrl
+      char* old_mode = g_settings_get_string(setting, "mode");
+      char* old_pac_url = g_settings_get_string(setting, "autoconfig-url");
+      if (strcmp(old_mode, "auto") != 0 || old_pac_url == NULL || strcmp(old_pac_url, pacUrl) != 0 ) {
```

Comment:
Does ``g_settings_get_string`` return an empty string if a key is undefined, i.e. could ``old_mode`` ever be NULL here? If so, do we need to account for that before passing it to ``strcmp``? Just confirming.